localstack/services/lambda_/event_source_listeners/utils.py

```diff
@@ -21,8 +23,14 @@ def filter_stream_records(records, filters: list[FilterCriteria]):
     for record in records:
         for filter in filters:
             for rule in filter["Filters"]:
-                filter_pattern: dict[str, any] = json.loads(rule["Pattern"])
-                if does_match_event(filter_pattern, record):
+                if config.EVENT_RULE_ENGINE == "java":
+                    event_str = json.dumps(record)
+                    event_pattern_str = rule["Pattern"]
+                    match_result = matches_rule(event_str, event_pattern_str)
+                else:
+                    filter_pattern: dict[str, any] = json.loads(rule["Pattern"])
+                    match_result = does_match_event(filter_pattern, record)
+                if match_result:
```

Comment:
Could we unify the interfaces, so we do not have to have a config switch before we invoke the method, but within the method? I see this `if config.EVENT_RULE_ENGINE` change a lot, but it really should be hidden from the users of the method, what do you think? Similar to the method in the ext call, we should have a unified interface for both.

Reply:
Yes, I think unifying should be the goal. Unfortunately, our implementations (see https://github.com/localstack/localstack/pull/10599) behave so differently that we first need to fix cross-implementation regressions before we can have a single unified method. Therefore, I just added an opt-in event ruler for each existing implementation (i.e., the else depends on the context) and pushed unification into a follow-up. Or do you think we should add such a helper method anyway for each implementation?

Reply:
As discussed with Daniel, let's skip this until we can unify the different implementations.